### ISSUE TITLE
Cluster backup endpoint should indicate failure

### ISSUE THREAD

**Issue by titanous** [flynn/flynn#2597]
The workflow should probably look like:
1. Create backup resource.
2. Start streaming backup from separate endpoint.
3. Check backup info endpoint for expected SHA and any error logs.

/cc @jvatic

**Comment by jvatic** [comment]
Here's my thoughts:

- Add a backups table (id, created_at, completed_at, sha, size, data)
- `POST /backups` saves a db record (id, created_at), starts job to generate tar and save into db (completed_at, sha, size, data)
- `GET /backups/:backups_id` returns backup record with `Accept: application/json`, otherwise streams backup (waiting for the job to complete if it hasn't)
- `GET /backup` does the same as the above endpoint, except assumes the latest backup and creates one if it doesn't exist for backwards compatibility (this endpoint can eventually be removed once API versioning is in place)
- `DELETE /backups/:backups_id` removes the backup from the db (I think in this case deleting the record instead of soft delete makes the most sense as backups can be rather large)
- Backup script should ensure records in backups table are not part of the postgres dump, though `pg_dumpall` doesn't appear to support the `--exclude-table-data` option so this could be an issue (maybe just store the tar data someplace else?)

**Comment by titanous** [comment]
The data should not be saved in the database. For now lets have the GET endpoint trigger the jobs and stream inline directly to the client (we can do something more sophisticated later). The database should store the status of the backup as well as the error log.

**Comment by titanous** [comment]
Flynn is unmaintained and our infrastructure will shut down on June 1, 2021. See [the README](https://github.com/flynn/flynn#flynn-is-unmaintained) for details.
